extensions/amp-story/1.0/test/test-amp-story.js

```diff
@@ -1198,6 +1199,245 @@ describes.realWin(
         });
       });
 
+      describe('amp-story-subscriptions navigation', () => {
+        let subscriptionsEl;
+        let storeService;
+
+        beforeEach(async () => {
+          toggleExperiment(win, 'amp-story-subscriptions', true);
+
+          storeService = new AmpStoryStoreService(win);
+          env.sandbox.stub(Services, 'storyStoreService').returns(storeService);
+
+          await createStoryWithPages(4, [
+            'cover',
+            'page-1',
+            'page-2',
+            'page-3',
+          ]);
+          subscriptionsEl = win.document.createElement(
+            'amp-story-subscriptions'
+          );
+          story.element.appendChild(subscriptionsEl);
+
+          // buildCallback() could be omitted as long as createStoryWithPages()
+          // is called because the getImpl() call inside it.
+          await story.layoutCallback();
+        });
+
+        describe('unknown subscription state', () => {
+          beforeEach(async () => {
+            // Block access with pending subscriptions state.
+            story.activePage_.element.dispatchEvent(
+              new MouseEvent('click', {clientX: 200})
+            );
+            story.activePage_.element.dispatchEvent(
+              new MouseEvent('click', {clientX: 200})
+            );
+            // Try to navigate to the first paywall page with state unknown so stays on current page.
+            expect(story.activePage_.element.id).to.equal('page-1');
+          });
+
+          it('should continue navigating to locked pages after the subscription state gets resolved to granted', async () => {
+            // Once grant status resolves to true, goes back to normal story experiences.
+            storeService.dispatch(
+              Action.TOGGLE_SUBSCRIPTIONS_STATE,
+              SubscriptionsState.GRANTED
+            );
+            await nextTick();
+            expect(story.activePage_.element.id).to.equal('page-2');
+
+            story.activePage_.element.dispatchEvent(
+              new MouseEvent('click', {clientX: 200})
+            );
+            expect(story.activePage_.element.id).to.equal('page-3');
+          });
+
+          it('should block after predefined delay after the subscription state gets resolved to blocked', async () => {
+            // Once grant status resolves to false, blocks the access to all locked pages and show the paywall.
+            // Note: would still go to the paywall page, e.g. page-2.
+            const dispatchSpy = env.sandbox.spy(storeService, 'dispatch');
```

Comment:
nit: we dont really need spy, we can get this property i.e. `expect(storeService.get(Property.SUBSCRIPTION)).to.equal(SubscriptionState.XXX)`